src/sage/modules/free_quadratic_module_integer_symmetric.py

```diff
@@ -1527,6 +1528,105 @@ def short_vectors(self, n, **kwargs):
         short = q.short_vector_list_up_to_length(n, *kwargs)
         return [[self(v * self.basis_matrix()) for v in L] for L in short]
 
+    def _fplll_enumerate(self, target=None):
+        r"""
+        Internal helper method to invoke the fplll enumeration routines.
+
+        EXAMPLES::
+
+            sage: L = IntegralLattice('A4')
+            sage: t = vector([1.2, -3/11, 5.5, -9.1])
+            sage: short = L.enumerate_short_vectors()   # implicit doctest
+            sage: [next(short) for _ in range(10)]
+            [(0, 0, 0, 1), (0, 0, 1, 1), (0, 1, 1, 1), (1, 1, 1, 1), (0, 0, 1, 0),
+             (1, 1, 1, 0), (0, 1, 1, 0), (0, 1, 0, 0), (1, 1, 0, 0), (1, 0, 0, 0)]
+            sage: close = L.enumerate_close_vectors(t)  # implicit doctest
+            sage: [next(close) for _ in range(10)]
+            [(1, 0, 6, -9), (1, -1, 5, -9), (2, 0, 6, -9), (1, 0, 5, -9), (1, -1, 5, -10),
+             (2, 1, 6, -9), (1, 0, 5, -10), (2, 0, 5, -9), (1, 0, 6, -8), (1, -1, 6, -9)]
+        """
+        L = self.LLL()
+        dim = L.dimension()
+        gram = L.gram_matrix()
+        basis = L.basis_matrix()
+
+        import fpylll
+        gmat = fpylll.IntegerMatrix(dim, dim)
+        for i in range(dim):
+            for j in range(dim):
+                gmat[i,j] = gram[i,j]
+        gso = fpylll.GSO.Mat(gmat, gram=True)
+        ok = gso.update_gso()
+        assert ok
+
+        coord = None
+        if target is not None:
+            coord = basis.solve_left(target)
+            Mu = 1 + matrix([gso.get_mu(i,j) for j in range(dim)] for i in range(dim))
```

Comment:
There's also a `to_matrix()` method, so something like:

```py
Mu = 1 + matrix(gso.to_matrix())
```

might work

Reply:
This doesn't work for me https://sagecell.sagemath.org/?q=yduvls (10.3)

Reply:
oh for the MatGSO type it's the property `.gso.G`

```
Mu = 1 + matrix(gso.G)
```

https://sagecell.sagemath.org/?q=qifgtc

Reply:
Ahha yes. I think this should be added before it's merged, seems quicker :-)